TagHelperSamples/src/TagHelperSamples.Bootstrap/NavLinkTagHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.TagHelpers;
+using Microsoft.AspNet.Mvc.ViewFeatures;
+using Microsoft.AspNet.Razor.TagHelpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TagHelperSamples.Bootstrap
+{
+    public class NavLinkTagHelper : AnchorTagHelper
+    {
+        public NavLinkTagHelper(IHtmlGenerator generator)
+            : base(generator)
+        {
+        }
+
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
```

Comment:
I wonder if it would be better to use ProcessAsync and the async await pattern here since you are calling output.GetChildContentAsync()